app/views/groups/show.html.erb

```diff
@@ -41,42 +46,47 @@
                         method: :post %>
           </p>
         <% end %>
+      <% elsif @cud.instructor %>
+        <p>
+          <%= link_to "Kick Member",
+                      leave_course_assessment_group_path(@course, @assessment, @group, member_id: cud.id),
+                      method: :post %>
+        </p>
       <% end %>
     </li>
   <% end %>
 </ul>
 
-<% if @group.is_member(@aud) or @cud.instructor then %>
-  <% if @group.assessment_user_data.size < @assessment.group_size then %>
-    <h2>Invite Another Student to Join This Group</h2>
+<% if user_can_edit_group? %>
+  <% if @group.assessment_user_data.size < @assessment.group_size %>
+    <h4>Invite Another Student to Join This Group</h4>
     <%= form_tag add_course_assessment_group_path(@course, @assessment, @group) do %>
       <div class="form-group">
         <label class="control-label" for="member_email">Partner Email</label>
         <%= text_field_tag :member_email, nil, class: "form-control", autocomplete: "email" %>
-        <p class="help-block">The email of your desired partner. They will need to confirm their membership.</p>
+        <% if @assessment.allow_student_assign_group %>
+          <p class="help-block">The email of your desired partner. They will need to confirm their membership.</p>
+        <% else %>
+          <%# Instructor adding a student %>
+          <p class="help-block">The email of your desired partner. Their membership will be automatically confirmed.</p>
+        <% end %>
       </div>
 
       <%= submit_tag "Invite to Group", class: "btn btn-primary" %>
     <% end %>
+  <% else %>
+    <p>No more members can be added as the group is already at the maximum size.</p>
   <% end %>
+<% elsif @group.is_member(@aud) && !@assessment.allow_student_assign_group %>
+  <p>
+    You are not allowed to add members to your group for this assessment.
+    Contact your instructor for group assignment.
+  </p>
 <% end %>
-
-<% if false and @grouplessCUDs then # disabled while we wonder if students should be trusted %>
-  <h2>Invite Another Student to Join This Group</h2>
-  <ul class="small gray-box">
-    <% @grouplessCUDs.each do |cud| %>
-      <% if cud.id != @cud.id && cud.student? then %>
-        <li><b><%= cud.email %></b>
-          (<%= link_to "Invite to Group", add_course_assessment_group_path(@course, @assessment, @group, member_id: cud.id),
-                       method: :post %>)
-        </li>
-      <% end %>
-    <% end %>
-  </ul>
-<% end %>
+<%# else: student assignment is allowed, but user is not a member of the group yet. We don't have to display anything %>
 
 <div style="margin-top: 1em">
-  <% if @group.is_member(@aud) then %>
+  <% if aud.group_id == id && (@assessment.allow_student_assign_group || @cud.instructor) %>
```

Comment:
Getting an error here which does not exist on the master branch. I'm unable to create a new group or edit the group.

To replicate:
- Change the group limit of an existing assessment
- Create a group and invite one user
- Click create group

Reply:
Fixed, mb